### PR TITLE
Introduce categorization

### PR DESCRIPTION
**Pull request by adpi2**

### CODE REVIEW

File: core/shared/src/test/scala/scaladex/core/model/CategoryTests.scala [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+import org.scalatest.funspec.AnyFunSpec
+import org.scalatest.matchers.should.Matchers
+
+class CategoryTests extends AnyFunSpec with Matchers {
+  it("should compute category label and title") {
+    Category.AlgorithmsAndDataStructures.label shouldBe "algorithms-and-data-structures"
+    Category.AlgorithmsAndDataStructures.title shouldBe "Algorithms and Data Structures"
+  }
+
+  Category.all.map(_.label).foreach(println)
```
Comment (mlachkar) [inline core/shared/src/test/scala/scaladex/core/model/CategoryTests.scala:12]: maybe remove this line

File: infra/src/main/resources/migrations/V3__init_category.sql [hunk 2]
```diff
@@ -0,0 +1,1246 @@
+ALTER TABLE project_settings ADD COLUMN category VARCHAR(63);
```
Comment (mlachkar) [inline infra/src/main/resources/migrations/V3__init_category.sql:1]: Awsome !!

File: infra/src/main/scala/scaladex/infra/util/DoobieUtils.scala [hunk 3]
```diff
@@ -147,11 +146,13 @@ object DoobieUtils {
       Meta[String].timap(fromJson[List[License]](_).get.toSet)(toJson(_))
     implicit val resolverMeta: Meta[Resolver] =
       Meta[String].timap(Resolver.from(_).get)(_.name)
-    implicit val instantMeta: doobie.Meta[Instant] = doobie.postgres.implicits.JavaTimeInstantMeta
+    implicit val instantMeta: Meta[Instant] = doobie.postgres.implicits.JavaTimeInstantMeta
 
-    implicit val projectReferenceRead: doobie.Read[Project.Reference] =
+    implicit val categoryRead: Meta[Category] = Meta[String].timap(Category.byLabel)(_.label)
```
Comment (mlachkar) [inline infra/src/main/scala/scaladex/infra/util/DoobieUtils.scala:151]:
```suggestion
    implicit val categoryMeta: Meta[Category] = Meta[String].timap(Category.byLabel)(_.label)
```

**Review by mlachkar — APPROVED** [review]

File: infra/src/main/scala/scaladex/infra/util/DoobieUtils.scala (hunk 3 above)
Comment (mlachkar) [inline infra/src/main/scala/scaladex/infra/util/DoobieUtils.scala:151]: it's weird there is not `.get ` in `Category.byLabel`

Comment (adpi2) [reply infra/src/main/scala/scaladex/infra/util/DoobieUtils.scala:151]: there is an `apply` method on `Map`

**Pull request merged**